src/components/VaultSummary.tsx

```diff
@@ -0,0 +1,51 @@
+import { useMemo } from 'react';
+import { useVaultStore } from 'store/vaults';
+import { useAtomValue } from 'jotai';
+import { displayFunctionsAtom } from 'store/app';
+import type { VaultKey } from 'store/vaults';
+
+type Props = {
+  id: VaultKey;
```

Comment:
IIUC, `VaultKey` is a composite key of [manager's id in the factory, vault's id within the manager]. 

Since that's the key of `vaults` and `errors` maps, shouldn't this be,
```suggestion
  key: VaultKet;
```
It overlaps with React's `key` property but not in any problematic way.

Reply:
The overlap is problematic, trying to access `key` as a prop returns undefined. Just changed to `vaultKey` since it's better than `id`

Reply:
> trying to access key as a prop returns undefined

TIL!